File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit =0;
        int current = prices[0];
        for(int  i =0 ; i< prices.size(); i++){
            current = min(current , prices[i]);
            int profit = prices[i] - current;
            maxprofit = max(profit , maxprofit);
        }
         return maxprofit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit = 0
        current = prices[0]
        for i in range(len(prices)):
            current = min(current , prices[i])
            profit = prices[i] - current
            maxprofit =max(profit , maxprofit)


        return maxprofit